src/cobra/util/context.py

```diff
@@ -49,10 +71,24 @@ def get_context(obj):
     return None
 
 
-def resettable(f):
-    """A decorator to simplify the context management of simple object
-    attributes. Gets the value of the attribute prior to setting it, and stores
-    a function to set the value to the old value in the HistoryManager.
+def resettable(f: Callable[[Any], Any]):
```

Comment:
`f` is a terrible argument name.

```suggestion
def resettable(f: Callable[[Any], Any]) -> Callable[[Any], Any]:
```

Reply:
Will change it.